decidim/decidim
# Provide bulk action answers to proposals through answers templates

#### :tophat: What? Why?
This pull request introduces the ability for administrators to perform bulk action answers on multiple proposals using template answers. This feature streamlines the process of responding to numerous proposals, enhancing efficiency and ensuring consistent communication.

#### NOTES
- This PR also fixes a bug within the model `ProposalState`. This model had a `before_action :save` hook that generated always an automatic token, regardless of the current value of the token. This behavior is ok when you create a new state and you want the token to be generated automatically but it's a problem when you want to control the actual value of the token.
- The solution is just use the token passed as a param if it exists and, if not, generate an automatic one.
- See the line 40 on the `ProposalState` file. Also tests have been fixed (and completed).

#### :pushpin: Related Issues
- Fixes #12359

### Testing
1. Log in as an administrator.
2. Ensure there are multiple proposals in the system.
3. Select multiple proposals.
4. Verify that the "Change status and answer" option is available in the bulk actions menu if at least one of the selected proposals allows an answer.
5. Apply a template answer to the selected proposals.
6. Verify that the proposals are updated with the template answer.
7. Verify that the "Change status and answer" option is not available in the bulk actions menu if none of the selected proposals allow an answer.

### :camera: Screenshots
![299938445-1bc4b4f6-baf9-4035-a5d7-ce01d11b818f-1](https://github.com/decidim/decidim/assets/60363870/b542bb5f-fced-4dec-b0c5-9f837f30d99d)
![299940025-a5a2af92-69ab-4679-a11e-a8462310e5ef-1](https://github.com/decidim/decidim/assets/60363870/225226f6-a6df-4dc6-92cd-04e7d93df9bf)
![Screenshot 2024-07-03 at 15 53 31](https://github.com/decidim/decidim/assets/60363870/34c60d9a-5d13-4296-8e2f-1c00a5861675)

:hearts: Thank you!
